flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/filesystem/FileSystemTableSink.java

```diff
@@ -343,7 +343,10 @@ private RowDataPartitionComputer partitionComputer() {
             @Override
             public DynamicTableSource.DataStructureConverter createDataStructureConverter(
                     DataType producedDataType) {
-                throw new TableException("Compaction reader not support DataStructure converter.");
+                // This method cannot be implemented without changing the
+                // DynamicTableSink.DataStructureConverter interface
+                throw new UnsupportedOperationException(
```

Comment:
Any idea why this exception was a `TableException` before?

Reply:
I don't, but the reason for this exception to exist is very clear to me: it's a problem with the design of `DataStructureConverter` propagation to the connectors, that doesn't allow the sink to create a converter which provides the `toInternal` logic. Hence, I think it's better to use `UnsupportedOperationException`.